DragaliaAPI/Controllers/Dragalia/DungeonRecordController.cs

```diff
@@ -92,38 +124,42 @@ public async Task<DragaliaResult> Record(DungeonRecordRecordRequest request)
                                     factor = 0,
                                 }
                         ),
-                        first_clear_set = new List<AtgenFirstClearSet>()
-                        {
-                            new()
+                        first_clear_set = clearedMissions[0]
+                            ? new List<AtgenFirstClearSet>()
                             {
-                                type = (int)EntityTypes.Wyrmite,
-                                id = 0,
-                                quantity = 5
+                                new()
+                                {
+                                    type = (int)EntityTypes.Wyrmite,
+                                    id = 0,
+                                    quantity = 5
+                                }
                             }
-                        },
+                            : new List<AtgenFirstClearSet>(),
                         take_coin = 1000,
                         take_astral_item_quantity = 300,
-                        missions_clear_set = Enumerable
-                            .Range(1, 3)
+                        missions_clear_set = clearedMissions
+                            .Where((x, index) => (index is > 0 and < 4) && x)
```

Comment:
is limiting the index necessary here?

Reply:
I think so, otherwise it'll include first clear bonus and all mission cleared bonus into the list as well which might cause problems?

Reply:
oh right, I misunderstood. Maybe it would be better to keep those variables out of the array in that case? seeing as they're used for different things

Reply:
Oh I'll do that then ye, I don't think it should mess with the logic of all missions cleared either since it's impossible for a first clear mission to be the last and only mission necessary to fulfill it